scripts/benchmark_test.sh

```diff
@@ -13,53 +13,100 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# This script runs a benchmark on a locally started etcd server
+# This utility script helps to set up the test environment for benchmark tests and runs the provided tests sequentially.
 
 set -euo pipefail
 
 source ./scripts/test_lib.sh
 
-COMMON_BENCHMARK_FLAGS="--report-perfdash"
+declare -A BENCHMARK_FLAGS
+BENCHMARKS_TO_RUN=()
+COMMON_BENCHMARK_FLAGS=(--report-perfdash)
 
-if [[ $# -lt 1 ]]; then
-  echo "Usage: $0 <benchmark-name> [tester args...]"
-  exit 1
-fi
+parse_args() {
+  for arg in "$@"; do
+    if [[ "$arg" == *:* ]]; then
+      key="${arg%%:*}"
+      value="${arg#*:}"
+      BENCHMARKS_TO_RUN+=("$key")
+      BENCHMARK_FLAGS["$key"]="$value"
+    else
+      BENCHMARKS_TO_RUN+=("$arg")
+      BENCHMARK_FLAGS["$arg"]=""
+    fi
+  done
 
-BENCHMARK_NAME="$1"
-ARGS="${*:2}"
+  if [ ${#BENCHMARKS_TO_RUN[@]} -eq 0 ]; then
+    log_error "Usage: ./benchmark_test.sh 'benchmark-operation:\"test_arguments\"'"
+    log_error "Example: ./benchmark_test.sh 'put:\"--clients=1000\" range:\"--conns=100\"'"
+    exit 1
+  fi
+}
 
-echo "Starting the etcd server..."
+start_etcd() {
+  log_callout "Setup: Starting the etcd server..."
+  # Create a directory for etcd data under /tmp/etcd
+  mkdir -p /tmp/etcd
+  DATA_DIR=$(mktemp -d /tmp/etcd/data-XXXXXX)
+  ARTIFACTS_DIR="${ARTIFACTS:-./_artifacts}"
+  mkdir -p "$ARTIFACTS_DIR"
+  ETCD_LOG_FILE="${ARTIFACTS_DIR}/etcd-${bench}-$(date +%Y%m%d-%H%M%S).log"
```

Comment:
Thought: Should it be good to have the logs uploaded to the GCS bucket along with the perfash report?